### ISSUE TITLE
SIGSEGV in requests with LowCardinality(String) from datagrip

### ISSUE THREAD

**Issue by loyd** [ClickHouse/ClickHouse#30183]
v21.9.3.30

```
2021.10.14 15:31:32.463779 [ 29135 ] {} <Fatal> BaseDaemon: ########################################
2021.10.14 15:31:32.463886 [ 29135 ] {} <Fatal> BaseDaemon: (version 21.9.3.30 (official build), build id: 81039AF87A01EB4B61B52813A0E168AC9D46C3BC) (from thread 17432) (query_id: 940ce119-2823-4863-864f-d12251056bda) Received signal Segmentation fault (11)
2021.10.14 15:31:32.463918 [ 29135 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Unknown si_code.
2021.10.14 15:31:32.463982 [ 29135 ] {} <Fatal> BaseDaemon: Stack trace: 0x10507477 0x10505e78 0x1050488e 0x10fb3328 0x11319b61 0x11319254 0x117e49c1 0x10ae8ca7 0x10aee40f 0x10afb4eb 0x10afbb6c 0x10afbc09 0x118211fd 0x1181d9d1 0x1181ba2f 0x1181b7f9 0x10ee6fe2 0x11746f0c 0x1174a692 0x117be610 0x143748ef 0x1437637a 0x144a91f9 0x144a548a 0x7f4b7c7d6609 0x7f4b7c6ec293
2021.10.14 15:31:32.464057 [ 29135 ] {} <Fatal> BaseDaemon: 1. DB::ReverseIndex<unsigned long, DB::ColumnString>::insert(StringRef const&) @ 0x10507477 in /usr/bin/clickhouse
2021.10.14 15:31:32.464087 [ 29135 ] {} <Fatal> BaseDaemon: 2. COW<DB::IColumn>::mutable_ptr<DB::IColumn> DB::ColumnUnique<DB::ColumnString>::uniqueInsertRangeImpl<char8_t>(DB::IColumn const&, unsigned long, unsigned long, unsigned long, DB::ColumnVector<char8_t>::MutablePtr&&, DB::ReverseIndex<unsigned long, DB::ColumnString>*, unsigned long) @ 0x10505e78 in /usr/bin/clickhouse
2021.10.14 15:31:32.464108 [ 29135 ] {} <Fatal> BaseDaemon: 3. DB::ColumnUnique<DB::ColumnString>::uniqueInsertRangeFrom(DB::IColumn const&, unsigned long, unsigned long) @ 0x1050488e in /usr/bin/clickhouse
2021.10.14 15:31:32.464195 [ 29135 ] {} <Fatal> BaseDaemon: 4. DB::ColumnLowCardinality::insertRangeFrom(DB::IColumn const&, unsigned long, unsigned long) @ 0x10fb3328 in /usr/bin/clickhouse
2021.10.14 15:31:32.464232 [ 29135 ] {} <Fatal> BaseDaemon: 5. DB::BufferSink::insertIntoBuffer(DB::Block const&, DB::StorageBuffer::Buffer&) @ 0x11319b61 in /usr/bin/clickhouse
2021.10.14 15:31:32.464251 [ 29135 ] {} <Fatal> BaseDaemon: 6. DB::BufferSink::consume(DB::Chunk) @ 0x11319254 in /usr/bin/clickhouse
2021.10.14 15:31:32.464326 [ 29135 ] {} <Fatal> BaseDaemon: 7. DB::ISink::work() @ 0x117e49c1 in /usr/bin/clickhouse
2021.10.14 15:31:32.464360 [ 29135 ] {} <Fatal> BaseDaemon: 8. DB::PushingToSinkBlockOutputStream::write(DB::Block const&) @ 0x10ae8ca7 in /usr/bin/clickhouse
2021.10.14 15:31:32.464380 [ 29135 ] {} <Fatal> BaseDaemon: 9. DB::PushingToViewsBlockOutputStream::write(DB::Block const&) @ 0x10aee40f in /usr/bin/clickhouse
2021.10.14 15:31:32.464461 [ 29135 ] {} <Fatal> BaseDaemon: 10. DB::AddingDefaultBlockOutputStream::write(DB::Block const&) @ 0x10afb4eb in /usr/bin/clickhouse
2021.10.14 15:31:32.464479 [ 29135 ] {} <Fatal> BaseDaemon: 11. DB::SquashingBlockOutputStream::finalize() @ 0x10afbb6c in /usr/bin/clickhouse
2021.10.14 15:31:32.464495 [ 29135 ] {} <Fatal> BaseDaemon: 12. DB::SquashingBlockOutputStream::writeSuffix() @ 0x10afbc09 in /usr/bin/clickhouse
2021.10.14 15:31:32.464585 [ 29135 ] {} <Fatal> BaseDaemon: 13. ? @ 0x118211fd in /usr/bin/clickhouse
2021.10.14 15:31:32.464611 [ 29135 ] {} <Fatal> BaseDaemon: 14. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0x1181d9d1 in /usr/bin/clickhouse
2021.10.14 15:31:32.464677 [ 29135 ] {} <Fatal> BaseDaemon: 15. DB::PipelineExecutor::executeImpl(unsigned long) @ 0x1181ba2f in /usr/bin/clickhouse
2021.10.14 15:31:32.464709 [ 29135 ] {} <Fatal> BaseDaemon: 16. DB::PipelineExecutor::execute(unsigned long) @ 0x1181b7f9 in /usr/bin/clickhouse
2021.10.14 15:31:32.464741 [ 29135 ] {} <Fatal> BaseDaemon: 17. DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, std::__1::shared_ptr<DB::Context>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, std::__1::optional<DB::FormatSettings> const&, std::__1::function<void ()>) @ 0x10ee6fe2 in /usr/bin/clickhouse
2021.10.14 15:31:32.464831 [ 29135 ] {} <Fatal> BaseDaemon: 18. DB::HTTPHandler::processQuery(DB::HTTPServerRequest&, DB::HTMLForm&, DB::HTTPServerResponse&, DB::HTTPHandler::Output&, std::__1::optional<DB::CurrentThread::QueryScope>&) @ 0x11746f0c in /usr/bin/clickhouse
2021.10.14 15:31:32.464850 [ 29135 ] {} <Fatal> BaseDaemon: 19. DB::HTTPHandler::handleRequest(DB::HTTPServerRequest&, DB::HTTPServerResponse&) @ 0x1174a692 in /usr/bin/clickhouse
2021.10.14 15:31:32.464918 [ 29135 ] {} <Fatal> BaseDaemon: 20. DB::HTTPServerConnection::run() @ 0x117be610 in /usr/bin/clickhouse
2021.10.14 15:31:32.464946 [ 29135 ] {} <Fatal> BaseDaemon: 21. Poco::Net::TCPServerConnection::start() @ 0x143748ef in /usr/bin/clickhouse
2021.10.14 15:31:32.464961 [ 29135 ] {} <Fatal> BaseDaemon: 22. Poco::Net::TCPServerDispatcher::run() @ 0x1437637a in /usr/bin/clickhouse
2021.10.14 15:31:32.464987 [ 29135 ] {} <Fatal> BaseDaemon: 23. Poco::PooledThread::run() @ 0x144a91f9 in /usr/bin/clickhouse
2021.10.14 15:31:32.465055 [ 29135 ] {} <Fatal> BaseDaemon: 24. Poco::ThreadImpl::runnableEntry(void*) @ 0x144a548a in /usr/bin/clickhouse
2021.10.14 15:31:32.465084 [ 29135 ] {} <Fatal> BaseDaemon: 25. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
2021.10.14 15:31:32.465105 [ 29135 ] {} <Fatal> BaseDaemon: 26. __clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
2021.10.14 15:31:33.222497 [ 29135 ] {} <Fatal> BaseDaemon: Checksum of the binary: 01C4CC39E7C1C73FEDDD9FDB4397D6B8, integrity check passed.
2021.10.14 15:31:52.837436 [ 15841 ] {} <Fatal> Application: Child process was terminated by signal 11.
```

**Comment by alexey-milovidov** [comment]
@den-crane See `DB::BufferSink::consume`.

**Comment by alexey-milovidov** [comment]
Fixed in #29782.
